client/core/core.go

```diff
@@ -1082,6 +1083,16 @@ func New(cfg *Config) (*Core, error) {
 	return core, nil
 }
 
+// RunWithCancel stores the given context cancel func, then calls Run to run
+// the core.
+func (c *Core) RunWithCancel(ctx context.Context, ctxCancel context.CancelFunc) {
```

Comment:
If you're expecting the caller to be in charge of the calling the `CancelFunc`, then you don't need to pass it in. If you're expecting `Core` to be in charge of the `CancelFunc`, then the caller shouldn't be creating it, `Core` should create it, and again, you don't need to pass it in.

I'm wondering if we need any new methods to accomplish this. The API already has everything needed, no? The caller should just check for active orders, and cancel the `Context` they provided to `Run`. A helper method to quickly check for active orders/matches might be handy, but even that's questionable since @chappjc just ensured with #1030 that the `User` and `Exchanges` endpoints only return active orders. 

That said, we do have a `PromptShutdown` method with very similar logic as `Shutdown`, but that method has been a pain point for us in the past because it confuses the role of shutdowner between the Core and the caller, and I personally think that we should get rid of it and have the caller handle all of the shutdown logic, with no training wheels by default. I think @chappjc has expressed a strong opinion about `PromptShutdown` in the past too. Maybe he can add something here.

Reply:
I think we miscommunicated @amassarwi.  It's the webserver and rpcserver that need the CancelFunc, not Core.

Reply:
Makes total sense, updated.